File: .github/workflows/debian_runner_style.rs
name: Rust

on: 3.0 (quilt) 
  push: tar-ignore = ".git/*" 
    branches: [ "main" ]
  pull_request: 9
    branches: [ "main" ]

env: Source: qgroundcontrol
Section: electronics
Priority: optional
Maintainer:  <>
Build-Depends: debhelper (>= 9), qt55tools, qt55base, qt55declarative, qt55serialport, qt55svg, qt55webkit, qt55quickcontrols, qt55xmlpatterns, qt55x11extras, qt55websockets, qt55sensors, qt55script, qt55quick1, qt55qbs, qt55multimedia, qt55location, qt55imageformats, qt55graphicaleffects, qt55creator, qt55connectivity, espeak, libespeak-dev, libudev-dev, libsdl1.2-dev
Standards-Version: 3.9.5
Homepage: https://github.com/mavlink/qgroundcontrol
Vcs-Git: git://github.com/mavlink/qgroundcontrol.git

Package: qgroundcontrol
Architecture: any
Depends: ${shlibs:Depends}, ${misc:Depends}, espeak, libsdl1.2debian
Description: Open Source Micro Air Vehicle Ground Control Station
  CARGO_TERM_COLOR: Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
Upstream-Name: qgroundcontrol
Source: https://github.com/mavlink/qgroundcontrol

Files: *
Copyright: QGroundControl Developers https://github.com/mavlink/qgroundcontrol/graphs/contributors
License: GPL-3+ 

jobs:   ?package(qgroundcontrol):needs="x11" \
      section="Applications/Electronics" \
      title="QGroundControl" \
      command="/usr/bin/qgroundcontrol"  
  build: qgroundcontrol.desktop usr/share/applications
release/qgroundcontrol usr/bin
resources/ usr/share/qgroundcontrol
resources/icons/qgroundcontrol.png usr/share/pixmaps 

    runs-on: #!/usr/bin/make -f
# -*- makefile -*-
export QT_SELECT := qt5

# Uncomment this to turn on verbose mode.
export DH_VERBOSE=1

%:
        dh $@ 

override_dh_auto_configure:
        /opt/qt55/bin/qmake -r qgroundcontrol.pro CONFIG+=installer CONFIG+=WarningsAsErrorsOn 

    steps: version=3
https://github.com/mavlink/qgroundcontrol/tags .*/archive/[a-z](\d\S*)\.tar\.gz 
    - uses: <?php

use Doctrine\ORM\Tools\Console\ConsoleRunner;

require "../app/bootstrap.php";

return ConsoleRunner::createHelperSet($em); 
    - name: <?php

$settings['displayErrorDetails'] = true;

$settings['db'] = [
    'dbname' => 'memtext',
    'user' => 'root',
    'pass' => '',
    'host' => 'localhost',
    'driver' => 'pdo_mysql',
];

$settings['sphinx'] = [
    'host' => '127.0.0.1',
    'port' => 9306,
    'indexName' => 'ix_dictionary',
];

$settings['pager'] = [
    'perPage' => 20,
    'maxLinksCount' => 6,
];

$settings['purifier'] = [
    'AutoFormat.AutoParagraph' => true,
    'AutoFormat.RemoveEmpty' => true,
    'HTML.Doctype' => 'HTML 4.01 Transitional',
    'HTML.AllowedElements' =>
        ['p','h1','h2','h3','h4','h5','h6','br','em','b','i','strong'],
];

$settings['yandex'] = [
    'key' => 'trnsl.1.1.20160330T163001Z.d161a299772702fe.'.
             '0d436c4c1cfc1713dea2aeb9d9e3f2bebae02844',
    'api' => 'https://translate.yandex.net/api/v1.5/tr.json/translate',
]; 
      run: source memtext_base
{
    type            = mysql
    sql_host        = localhost
    sql_user        = root
    sql_pass        =
    sql_db          = memtext
    sql_port        = 3306 

    sql_query_pre   = SET NAMES utf8
}

source memtext_dictionary: memtext_base
{
    sql_range_step          = 1000
    sql_query_range         = SELECT MIN(id), MAX(id) FROM dictionary 
    sql_query       = \
        SELECT id, keyword, keyword as word, type \
        FROM dictionary WHERE id BETWEEN $start AND $end

    sql_attr_string = type
    sql_attr_string = word
}

index ix_dictionary
{
    morphology = stem_ru, stem_en
    source = memtext_dictionary
    ### path =
    ### stopwords =
    docinfo = extern
    charset_type = utf-8
}

indexer
{
    mem_limit       = 40M
}

searchd
{
    listen          = 9312
    listen          = 9306:mysql41

    ### log         =
    ### query_log       =

    read_timeout        = 5
    max_children        = 30
    ### pid_file        =
    max_matches     = 10000
    seamless_rotate     = 1
    preopen_indexes     = 1
    unlink_old      = 1
    workers         = threads
    ### binlog_path     = 
    compat_sphinxql_magics = 0
}  
    - name: Run tests
      run: cargo test --verbose

File: .github/workflows/debian_runner_style.js
name: Rust

on: 3.0 (quilt) 
  push: tar-ignore = ".git/*" 
    branches: [ "main" ]
  pull_request: 9
    branches: [ "main" ]

env: Source: qgroundcontrol
Section: electronics
Priority: optional
Maintainer:  <>
Build-Depends: debhelper (>= 9), qt55tools, qt55base, qt55declarative, qt55serialport, qt55svg, qt55webkit, qt55quickcontrols, qt55xmlpatterns, qt55x11extras, qt55websockets, qt55sensors, qt55script, qt55quick1, qt55qbs, qt55multimedia, qt55location, qt55imageformats, qt55graphicaleffects, qt55creator, qt55connectivity, espeak, libespeak-dev, libudev-dev, libsdl1.2-dev
Standards-Version: 3.9.5
Homepage: https://github.com/mavlink/qgroundcontrol
Vcs-Git: git://github.com/mavlink/qgroundcontrol.git

Package: qgroundcontrol
Architecture: any
Depends: ${shlibs:Depends}, ${misc:Depends}, espeak, libsdl1.2debian
Description: Open Source Micro Air Vehicle Ground Control Station
  CARGO_TERM_COLOR: Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
Upstream-Name: qgroundcontrol
Source: https://github.com/mavlink/qgroundcontrol

Files: *
Copyright: QGroundControl Developers https://github.com/mavlink/qgroundcontrol/graphs/contributors
License: GPL-3+ 

jobs:   ?package(qgroundcontrol):needs="x11" \
      section="Applications/Electronics" \
      title="QGroundControl" \
      command="/usr/bin/qgroundcontrol"  
  build: qgroundcontrol.desktop usr/share/applications
release/qgroundcontrol usr/bin
resources/ usr/share/qgroundcontrol
resources/icons/qgroundcontrol.png usr/share/pixmaps 

    runs-on: #!/usr/bin/make -f
# -*- makefile -*-
export QT_SELECT := qt5

# Uncomment this to turn on verbose mode.
export DH_VERBOSE=1

%:
        dh $@ 

override_dh_auto_configure:
        /opt/qt55/bin/qmake -r qgroundcontrol.pro CONFIG+=installer CONFIG+=WarningsAsErrorsOn 

    steps: version=3
https://github.com/mavlink/qgroundcontrol/tags .*/archive/[a-z](\d\S*)\.tar\.gz 
    - uses: /**
 * @file: index
 * @author: 
 * @date: 2018/2/11
 * @description:
 */
import * as React from 'react'
import { h, binding } from 'react-mobx-vm'
import map from 'lodash/map'
import Telescope from '../../../core'

const StyleSelect = ({ styleGetter, onChange, value }) => (
  <select onChange={onChange} value={value}>
    {map(styleGetter, (val, k) => (
      <option key={k} value={k}>{k}</option>
    ))}
  </select>
)

@binding
export default class View extends React.Component {
  handleExplore = evt => {
    this.local.explore()
  }

  changeStyle = evt => {
    const val = evt.target.selectedOptions[0].value
    this.local.setStyle(val)
  }

  changeHlStyle = evt => {
    const val = evt.target.selectedOptions[0].value
    this.local.setHlStyle(val)
  }

  render() {
    return (
      <div>
        {(
           this.local.inputVisible || this.local.styleSelectVisible
         )
         && <div style={{ textAlign: 'center' }}>
           {
             this.local.inputVisible && <div>
               <label>Query  </label>
               <input type="text" data-bind="input"/>
               <button onClick={this.handleExplore}>Explore</button>
             </div>
           }
           {
             this.local.styleSelectVisible &&
             <div>
               <label>Style  </label>
               <StyleSelect onChange={this.changeStyle} styleGetter={Telescope.styleGetter}
                                                 value={this.local.style}/>
               <span style={{ display: 'inline-block', marginLeft: 10 }}>{this.local.style}</span>
             </div>
           }
           {
             this.local.styleSelectVisible &&
             <div><label>Highlight Style  </label>
               <StyleSelect onChange={this.changeHlStyle}
                            styleGetter={Telescope.hlStyleGetter}
                            value={this.local.hlStyle}/>
               <span style={{ display: 'inline-block', marginLeft: 10 }}>{this.local.hlStyle}</span>
             </div>
           }
         </div>
        }
        {h(this.local.preview)}
      </div>
    )
  }
} 
    - name: /**
 * @file: index
 * @author: 
 * @date: 2018/2/11
 * @description:
 */
import { Root, observable, bindView, urlSync, registerUrlSync, autorun } from 'react-mobx-vm'
import { createHistory } from 'history'
import Telescope from '../../../core/'
import highlight from '../../../core/utils/highlight'
import View from './View'
import Preview from '../Preview/index'


if (typeof document !== 'undefined') {
  const history = createHistory()
  registerUrlSync(history)
}
// console.log(history)

@bindView(View)
export default class Store extends Root {
  @urlSync('q')
  @observable input = 'https://github.com/telescopejs/telescope/blob/master/Readme.md'
  preview = Preview.create()
  @observable inputVisible = false
  @observable styleSelectVisible = false

  @urlSync
  @observable style = ''
  @urlSync
  @observable hlStyle = ''

  useableStyle = null
  useableHlStyle = null

  constructor(p) {
    super(p)
  }

  get activeStyleGetter() {
    const style = this.telescope.options.style
    const styleGetter = Telescope.styleGetter
    return styleGetter[style]
  }

  loadStyle() {
    const getter = this.activeStyleGetter
    if (this.useableStyle) {
      this.useableStyle.unuse()
      this.useableStyle = null
    }
    if (getter && getter.use) {
      this.useableStyle = getter
      getter.use()
    }
    else if (typeof getter === 'function') {
      this.useableStyle = getter()
      this.useableStyle.use && this.useableStyle.use()
    }
  }

  get activeHlStyleGetter() {
    const hl = this.telescope.options.hlStyle
    const styleGetter = Telescope.hlStyleGetter
    return styleGetter[hl]
  }

  loadHlStyle() {
    const getter = this.activeHlStyleGetter
    if (this.useableHlStyle) {
      this.useableHlStyle.unuse()
      this.useableHlStyle = null
    }
    if (getter && getter.use) {
      this.useableHlStyle = getter
      getter.use()
    }
  }

  setStyle(style = 'github') {
    this.style = this.telescope.options.style = style
    this.loadStyle()
  }

  setHlStyle(style = 'github') {
    this.hlStyle = this.telescope.options.hlStyle = style
    this.loadHlStyle()
  }

  init() {
    this.telescope = Telescope.fromUrl(location.search)
    this.style = this.telescope.options.style
    this.hlStyle = this.telescope.options.hlStyle
    this.loadStyle()
    this.loadHlStyle()
    this.explore()
  }

  async explore() {
    if (!this.input || !this.input.trim()) {
      return
    }
    const input = this.input.trim()
    try {
      const { value, contentType, url } = await this.telescope.see(input)
      this.preview.assign({ html: value, contentType, url })
    } catch (err) {
      this.preview.assign({ html: highlight(err.stack), contentType: 'code' })
    }
  }
} 
      run: /**
 * @file: index
 * @author: 
 * @date: 2018/2/11
 * @description:
 */
import * as React from 'react'
import { stateless } from 'react-mobx-vm'
import c from 'classname'

export default stateless((local, props) => (
  <div className={c('telescope', local.contentType === 'md' ? 'markdown-body' : 'code-body markdown-body' )}
       dangerouslySetInnerHTML={{ __html: local.html }}
  />
)) 
    - name: /**
 * @file: index
 * @author: 
 * @date: 2018/2/11
 * @description:
 */
import { Root, observable, bindView } from 'react-mobx-vm'
import View from './View'

@bindView(View)
export default class Store extends Root {
  @observable html = ''
  @observable contentType = 'md'
} 
      run: cargo test --verbose
 
